rag_experiment_accelerator/nlp/preprocess.py

```diff
@@ -1,25 +1,20 @@
 import re
-from string import punctuation
 
-import spacy
+from abc import ABC
+from string import punctuation
+from typing import Optional
 
 from rag_experiment_accelerator.utils.logging import get_logger
 
 logger = get_logger(__name__)
 
 
-class Preprocess:
-    def __init__(self):
-        try:
-            self.nlp = spacy.load("en_core_web_lg")
-        except OSError:
-            logger.info(
-                f'Downloading spacy language model: {"en_core_web_lg"}'
-            )
-            from spacy.cli import download
+class Preprocess(ABC):
```

Comment:
Why are you making Preprocess an abstract base class?

Are we saying Preprocess should not be instantiated?

Reply:
you are correct, this shouln't be an abstract class,
originially it was changed by Omri to an abstract and there was another class call EmptyPreprocess which basically did nothing,
I've decided to move that logic here, and allow others to override this behavior.

Reply:
you are correct, this shouln't be an abstract class,
originially it was changed by Omri to an abstract and there was another class call EmptyPreprocess which basically did nothing,
I've decided to move that logic here, and allow others to override this behavior.

Reply:
Just to share the original approach: Preprocessing can be done in different ways, and by having an abstract class, one can implement different types of preprocessing while maintaining a consistent interface.

Replacing spacy with am empty preprocessor does change the behavior, but IMHO there is no nees for actions like stopwords removal which are more common in traditional NLP pipelines.

Reply:
If the abstract class is over-engineering this, we can definitely remove it.

Reply:
My point was really that it's not abstract (it has concrete implementation, and we're creating instances of it). Using it as a class from which we derive other classes is fine.

Reply:
The initial version had an abstract `Preprocess` class with two concrete implementations: `EmptyPreprocessor` and `SpacyPreprocessor`. The code was changed based on the feedback given here.

Reply:
after today's review meeting, we have decided to drop class inheritance for code simplicity and add config parameter for enabling/disabling preprocess.